tests/semantic_test.py

```diff
@@ -0,0 +1,561 @@
+from pathlib import Path
+from itertools import product
+from typing import Dict, Tuple
+
+import pytest
+
+from .compile_asserts import assert_teal_as_expected
+from .semantic_asserts import (
+    algod_with_assertion,
+    get_blackbox_scenario_components,
+    e2e_pyteal,
+    mode_has_assertion,
+)
+
+from algosdk.testing.dryrun import Helper as DryRunHelper
+from algosdk.testing.teal_blackbox import (
+    DryRunAssertionType as DRA,
+    SequenceAssertion,
+    csv_from_dryruns,
+    dryrun_assert,
+    lightly_encode_args,
+    lightly_encode_output,
+    scratch_encode,
+)
+
+from pyteal import *
+
+# TODO: get tests working on Circle and set this to True
+SEMANTIC_TESTING = False
```

Comment:
Currently not actually running the dry-runs on github. Worth discussing how to move forward o this.